src/mscorlib/src/System/Buffer.cs

```diff
@@ -507,8 +507,9 @@ internal unsafe static void Memmove(byte* dest, byte* src, nuint len)
             *(nuint*)dest = *(nuint*)src;
 
             // byte offset at which we're copying
-            int offset = sizeof(nuint) - ((int)dest % sizeof(nuint)); // maps 0 => 8, 1 => 7, ... 8 => 8 for 64-bit
```

Comment:
`(int)dest` will be negative if dest is, say, `(long)int.MaxValue + 1`. Also using `uint` requires you to cast *everything* to `uint`, e.g.

```cs
uint offset = (uint)sizeof(nuint) - ((uint)dest % (uint)sizeof(nuint));
```

In the end I found it was easier to just work with ints and then cast to `nuint`.